guava/src/main/java/com/fasterxml/jackson/datatype/guava/deser/GuavaCollectionDeserializer.java

```diff
@@ -132,15 +132,19 @@ public Object deserializeWithType(JsonParser p, DeserializationContext ctxt,
     public T deserialize(JsonParser p, DeserializationContext ctxt)
             throws IOException
     {
-        // Should usually point to START_ARRAY
-        if (p.isExpectedStartArrayToken()) {
-            return _deserializeContents(p, ctxt);
-        }
-        // But may support implicit arrays from single values?
-        if (ctxt.isEnabled(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY)) {
-            return _deserializeFromSingleValue(p, ctxt);
+        try {
+            // Should usually point to START_ARRAY
+            if (p.isExpectedStartArrayToken()) {
+                return _deserializeContents(p, ctxt);
+            }
+            // But may support implicit arrays from single values?
+            if (ctxt.isEnabled(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY)) {
+                return _deserializeFromSingleValue(p, ctxt);
+            }
+            return (T) ctxt.handleUnexpectedToken(_valueClass, p);
+        } catch (NullPointerException e) {
```

Comment:
No. We'd need to know why NPE was thrown and prevent that instead -- a unit test reproduction would be useful.

Reply:
I have added in unit test and also fix the logic of how to handle those NPE in https://github.com/FasterXML/jackson-datatypes-collections/pull/125/commits/3f3f61f66a43d0964693d74cd4aac3b513cdf7a1.